### PR TITLE
Require profile fields to belong to a group

### PR DESCRIPTION
**Pull request by djuber**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

As we move to make the profile fields option generally accessible (remove the feature flag guards), there is a behavior that no longer makes sense regarding ungrouped fields. I clarified with Jenny that these should not be possible to create (any custom field should belong to a profile field group).

Remove the empty group dropdown from the group select tag

Remove the optional: true from the belongs_to validation (raise a
validation error on save if the group is not provided)

Update test to remove belongs_to's optional setting

## Related Tickets & Documents

Internal slack discussion clarifying the intentions for this feature: 
https://forem-team.slack.com/archives/C02P0RUGS04/p1643750964238099?thread_ts=1643747780.470129&cid=C02P0RUGS04

## QA Instructions, Screenshots, Recordings

Enable the `profile_admin` feature flag at http://localhost:3000/admin/feature_flags/features 

Try to make a profile field with no group - you should receive a validation error

```ruby
 pf = ProfileField.new(label: "a label")
 pf.errors.messages
=> {:profile_field_group=>["must exist"]

 pf.save!
  TRANSACTION (1.9ms)  BEGIN
  ProfileField Exists? (1.9ms)  SELECT 1 AS one FROM "profile_fields" WHERE LOWER("profile_fields"."label") = LOWER($1) LIMIT $2  [["label", "a label"], ["LIMIT", 1]]
  TRANSACTION (1.1ms)  ROLLBACK
ActiveRecord::RecordInvalid: Validation failed: Profile field group must exist
```

Visit the admin page for profile groups http://localhost:3000/admin/customization/profile_fields

On main, there's a blank option in the group select dropdown:

empty-group-option

On this branch, that's gone:

no-empty-dropdown-option

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [x] I will share this change internally with the appropriate teams

## [optional] Are there any post deployment tasks we need to perform?

We may have invalid fields on forem's that have had this feature flag enabled. Those will still show (and can be edited to belong to a group or deleted, but can't be updated otherwise) in admin.

We still have the "ungrouped profile fields" partial, that should be removed once there aren't any concerns about existing invalid data. 

There's a `dependent: nulllify` condition in the ProfileFieldGroup model - should this be destroy (or is that the _other_ case where a field may have no group) or would that mean we want to keep the ungrouped profile fields as a catch-all to collect fields where the group was removed? 

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Review by msarit — APPROVED** [review]

**Comment by djuber** [comment]
I need to fix those test failures (they're related to this change, and the system worked).

**Review by citizen428 — APPROVED** [review]
This makes sense, the use-case for profile fields without groups (settings in disguise with a display property set to `false`) has been replaced by having proper `Settings::*` models.

File: spec/factories/profile_fields.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 FactoryBot.define do
   factory :profile_field do
-    profile_field_group
+    profile_field_group { FactoryBot.create(:profile_field_group) }
```
Comment (citizen428) [inline spec/factories/profile_fields.rb:3]: Maybe we can use `association` instead as outlined [in the documentation](https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#explicit-definition).

```suggestion
    association: :profile_field_group
```

**Review by citizen428 — APPROVED** [review]

**Comment by djuber** [comment]
> Will this cause existing profile fields to fail to be updated?

No - and it _also_ doesn't change the [nullify behavior](https://github.com/forem/forem/blob/main/app/models/profile_field_group.rb#L2) of profile field group destruction (it will be possible to have fields get into no group, but they will need to have a group assigned the next time they're edited). They won't show in profiles until they're assigned a group. An admin won't be able to mark "no group" on a field during an edit (the only path to that state would be group deletion).

That _does_ cause issues if we had an update script modify profile fields in the future (from ruby, with validations) since some fields might fail validation. Since there's not guaranteed to even be a group on the system, it doesn't feel warranted to create one just so we can have a place to put these (that would have a user facing effect of exposing previously hidden fields), for the same reason I'd be uncomfortable taking ProfileFieldGroup.first and moving unassigned fields there during this change.

**Review by jgaskins — APPROVED** [review]
Sounds reasonable, thanks @djuber!

**Comment by djuber** [comment]
The test cases skipped are for data updates which will be removed in #16396

**Pull request merged**